plio/views.py

```diff
@@ -114,14 +124,56 @@ def duplicate(self, request, uuid):
         return Response(self.get_serializer(plio).data)
 
     @action(methods=["get"], detail=True, permission_classes=[IsAuthenticated])
-    def unique_users(self, request, uuid):
-        """Returns the number of unique users who have watched the specified plio"""
-        session_queryset = Session.objects.filter(plio__uuid=uuid)
-        return Response(
-            session_queryset.aggregate(Count("user__id", distinct=True))[
-                "user__id__count"
-            ]
-        )
+    def download_data(self, request, uuid):
+        # return 404 if user cannot access the object
+        # else fetch the object
+        plio = self.get_object()
+
+        # handle draft plios
+        if plio.status == "draft":
+            return Response(
+                {"detail": "Data dumps are not available for draft plios"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        # holds the data dump
+        # data_dump = {}
+        data_dump_dir = f"/tmp/plio-{uuid}/{request.user.id}"
+        if os.path.exists(data_dump_dir):
+            shutil.rmtree(data_dump_dir)
+        os.makedirs(data_dump_dir)
+
+        # schema name to query in
+        schema_name = OrganizationTenantMiddleware().get_schema(self.request)
+
+        def save_query_results(query_method, filename):
+            cursor.execute(query_method(uuid, schema=schema_name))
+            columns = [col[0] for col in cursor.description]
+            df = pd.DataFrame(cursor.fetchall(), columns=columns)
+            df.to_csv(os.path.join(data_dump_dir, filename), index=False)
+
+        # create the individual dump files
+        with connection.cursor() as cursor:
+            save_query_results(get_sessions_dump_query, "sessions.csv")
+            save_query_results(get_responses_dump_query, "responses.csv")
+            save_query_results(get_plio_details_query, "plio-interaction-details.csv")
+            save_query_results(get_events_query, "events.csv")
+
+            df = pd.DataFrame(
+                [[plio.uuid, plio.name, plio.video.url]],
+                columns=["id", "name", "video"],
+            )
+            df.to_csv(os.path.join(data_dump_dir, "plio-meta-details.csv"), index=False)
+
+        # create the zip
+        shutil.make_archive(data_dump_dir, "zip", data_dump_dir)
+
+        # read the zip
+        zip_file = open(f"{data_dump_dir}.zip", "rb")
+
+        # create the response
+        response = FileResponse(zip_file, as_attachment=True)
```

Comment:
Shouldn't we clear the data dump stored in `tmp` directory before exiting from the function?